bokehjs/src/coffee/models/widgets/multiselect.coffee

```diff
@@ -39,15 +39,25 @@ export class MultiSelectView extends InputWidgetView
       if values[el.attr('value')]
         el.attr('selected', 'selected')
     )
+    # Note that some browser implementations might not reduce
+    # the number of visible options for size <= 3.
     @$el.find('select').attr('size', this.model.size)
 
   change_input: () ->
+    # Haven't checked if :focus selector works for IE <= 7
+    is_focused = @$el.find('select:focus').size()
     value = @$el.find('select').val()
     if value
       @model.value = value
     else
       @model.value = []
     super()
+    # Restore focus back to the <select> afterwards,
+    # so that even if python on_change callback is invoked,
+    # focus remains on <select> and one can seamlessly scroll
+    # up/down.
+    if is_focused
+      @$el.find('select').focus()
```

Comment:
any thoughts on the actual implementation to keep focus? If not I will merge tomorrow.